### PR TITLE
Remove validation when adding or updating consumer

### PR DESCRIPTION
**Pull request by scottf**
1. Do not require name or durable when adding or updating a consumer.

### CODE REVIEW

File: src/NATS.Client/JetStream/JetStreamManagement.cs [hunk 1]
```diff
@@ -86,7 +86,6 @@ public ConsumerInfo AddOrUpdateConsumer(string streamName, ConsumerConfiguration
         {
             Validator.ValidateStreamName(streamName, true);
             Validator.ValidateNotNull(config, nameof(config));
-            Validator.ValidateNotNull(config.Durable, nameof(config.Durable)); // durable name is required when creating consumers
```
Comment (Jarema) [inline src/NATS.Client/JetStream/JetStreamManagement.cs:89]: If client does not set durable name, it has to generate it on it's own, unless it uses the old ephemeral API, which I don't think is a good idea.

I don't see a change with generating new name - so it uses the old ephemeral API?

Comment (scottf) [reply src/NATS.Client/JetStream/JetStreamManagement.cs:89]: If neither name or durable are supplied, yes, it uses the old api. `CONSUMER.CREATE.%s`

Comment (scottf) [reply src/NATS.Client/JetStream/JetStreamManagement.cs:89]: I could add code here to generate a name so it uses the new api
`CONSUMER.CREATE.%s.%s` or if there is a filter `CONSUMER.CREATE.%s.%s.%s`

**Review by piotrpio — APPROVED** [review]
LGTM

**Pull request merged**